pkg/controller/internal/transportadapter/transportadapter_translator.go

```diff
@@ -242,6 +254,86 @@ func (t *transportAdapterTranslator) createSecretEnvFrom(
 	return envFrom
 }
 
+// createVolumes creates volumes from the MCPServer deployment spec
+func (t *transportAdapterTranslator) createVolumes(
+	deployment v1alpha1.MCPServerDeployment,
+) []corev1.Volume {
+	volumes := make([]corev1.Volume, 0)
+
+	// Add volumes from SecretRefs
+	for _, secretRef := range deployment.SecretRefs {
+		if secretRef.Name == "" {
+			continue
+		}
+		volumes = append(volumes, corev1.Volume{
+			Name: fmt.Sprintf("secret-%s", secretRef.Name),
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: secretRef.Name,
+				},
+			},
+		})
+	}
+
+	// Add volumes from ConfigMapRefs
+	for _, configMapRef := range deployment.ConfigMapRefs {
+		if configMapRef.Name == "" {
+			continue
+		}
+		volumes = append(volumes, corev1.Volume{
+			Name: fmt.Sprintf("configmap-%s", configMapRef.Name),
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: configMapRef.Name,
+					},
+				},
+			},
+		})
+	}
+
+	// Add custom volumes
+	volumes = append(volumes, deployment.Volumes...)
+
+	return volumes
+}
+
+// createVolumeMounts creates volume mounts from the MCPServer deployment spec
+func (t *transportAdapterTranslator) createVolumeMounts(
+	deployment v1alpha1.MCPServerDeployment,
+) []corev1.VolumeMount {
+	volumeMounts := make([]corev1.VolumeMount, 0)
+
+	// Add volume mounts from SecretRefs
+	for _, secretRef := range deployment.SecretRefs {
+		if secretRef.Name == "" {
+			continue
+		}
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      fmt.Sprintf("secret-%s", secretRef.Name),
+			MountPath: fmt.Sprintf("/secrets/%s", secretRef.Name),
+			ReadOnly:  true,
+		})
+	}
```

Comment:
same as above I don't think this is necessary as we already add the secret data as environment variables in the container via the `envFrom` stanza